src/ngraph/runtime/backend.hpp

```diff
@@ -52,6 +52,12 @@ namespace ngraph
                 make_primary_tensor_view(const ngraph::element::Type& element_type,
                                          const Shape& shape) = 0;
 
+            /// @brief Return a handle for a tensor for given mem on backend device
```

Comment:
do we want to allow users to use preallocated cuda mem for GPU, and similar for NNP.  For this PR, i can make it specific to cpu.

Reply:
I added @fengleitian and @yxlao to the PR to get their feedback on GPU and NNP backends

Reply:
Yeah, this wouldn't generalize to NNP. If we were to implement similar thing, it would be a `ArTensorId` instead of `void*`.

Another way to use it would be: instead of making `void* memory_pointer` as "use a pre-allocated NNP buffer", we treat it as "create a NNP tensor and copy the data from host `memory_pointer` to NNP". But we already have another APIs to do that.

Reply:
CUDA side should be able to use `void*`, but need to allocate with `cudaMalloc` and free with `cudaFree`.

Reply:
this PR is to enable that use-case in general. 

So we have 3 backends that can use this, and 4th one (NNP) cannot use this directly. For consistency, can we still "create a NNP tensor and copy the data from host memory_pointer to NNP" .  Or just proxy it to existing NNP APIs that do it.

Reply:
latest commit has GPU added, this allows user to specify cuda mem that they may manage themselves.

Reply:
I mean it will work on GPU, but I have some dumb questions:
1. Is the tensorviews itself unique, why we need memory to uniquely tag tensorviews? 
2. if user need to create memory, they first need to get information from tensorviews? like size of the memory? why not just get a pointer to the data?

Reply:
1. yes, tensorview itself is expected to be unique. This PR makes this more transparent.
2. some frameworks already have their own memory management, this API allows to just reuse existing pre-allocated mem as needed. Also, shape/type (size) is already known before creating tensorview.

Reply:
OK, that make sense. Thanks @ashokei